configure-ironic.sh

```diff
@@ -46,6 +51,36 @@ connection = mysql+pymysql://ironic:${MARIADB_PASSWORD}@localhost/ironic?charset
 http_url = http://${IRONIC_URL_HOST}:${HTTP_PORT}
 fast_track = ${IRONIC_FAST_TRACK}
 
+EOF
+
+if [ ! -z "$CACERT_FILE" ] || [ ! -z "$INSECURE" ]; then
+    crudini --merge /etc/ironic/ironic.conf <<EOF
+[inspector]
+endpoint_override = https://${IRONIC_URL_HOST}:5050
+power_off = ${INSPECTOR_POWER_OFF}
+$([ ! -z "$CACERT_FILE" ] && echo "cafile = $CACERT_FILE")
+$([ ! -z "$INSECURE" ] && echo "insecure = $INSECURE" || echo "insecure = false")
+# TODO(dtantsur): ipa-api-url should be populated by ironic itself, but it's
+# not, so working around here.
+# NOTE(dtantsur): keep inspection arguments synchronized with inspector.ipxe
+extra_kernel_params = console=ttyS0 ipa-insecure=1 ipa-inspector-collectors=default,extra-hardware,logs ipa-inspection-dhcp-all-interfaces=1 ipa-collect-lldp=1 ${INSPECTOR_EXTRA_ARGS}
+[pxe]
+pxe_append_params = nofb nomodeset vga=normal ipa-insecure=1
+[service_catalog]
+endpoint_override = https://${IRONIC_URL_HOST}:6385
+$([ ! -z "$CACERT_FILE" ] && echo "cafile = $CACERT_FILE")
+$([ ! -z "$INSECURE" ] && echo "insecure = $INSECURE" || echo "insecure = false")
+[ssl]
+$([ ! -z "$CACERT_FILE" ] && echo "cafile = $CACERT_FILE")
```

Comment:
Are you planning on client certificate validation? Won't work with the ramdisk, let's skip for now.

Reply:
ok, I'll remove the CA